cxx-qt-lib/src/qdatetime.rs

```diff
@@ -0,0 +1,198 @@
+    #[link_name = "cxxqt1$unique_ptr$qdatetime$get"]
+    fn unique_ptr_qdatetime_get(this: *const MaybeUninit<*mut c_void>) -> *const QDateTime;
+    #[link_name = "cxxqt1$unique_ptr$qdatetime$release"]
+    fn unique_ptr_qdatetime_release(this: *mut MaybeUninit<*mut c_void>) -> *mut QDateTime;
+    #[link_name = "cxxqt1$unique_ptr$qdatetime$drop"]
+    fn unique_ptr_qdatetime_drop(this: *mut MaybeUninit<*mut c_void>);
+}
+
+// Safety: TODO
+unsafe impl UniquePtrTarget for QDateTime {
+    #[doc(hidden)]
+    fn __typename(f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        f.write_str("QDateTime")
+    }
+
+    #[doc(hidden)]
+    fn __null() -> MaybeUninit<*mut c_void> {
+        let mut repr = MaybeUninit::uninit();
+        unsafe {
+            unique_ptr_qdatetime_null(&mut repr);
+        }
+        repr
+    }
+
+    #[doc(hidden)]
+    unsafe fn __raw(raw: *mut Self) -> MaybeUninit<*mut c_void> {
+        let mut repr = MaybeUninit::uninit();
+        unique_ptr_qdatetime_raw(&mut repr, raw);
+        repr
+    }
+
+    #[doc(hidden)]
+    unsafe fn __get(repr: MaybeUninit<*mut c_void>) -> *const Self {
+        unique_ptr_qdatetime_get(&repr)
+    }
+
+    #[doc(hidden)]
+    unsafe fn __release(mut repr: MaybeUninit<*mut c_void>) -> *mut Self {
+        unique_ptr_qdatetime_release(&mut repr)
+    }
+
+    #[doc(hidden)]
+    unsafe fn __drop(mut repr: MaybeUninit<*mut c_void>) {
+        unique_ptr_qdatetime_drop(&mut repr)
+    }
+}
+
+/// The Rust representation of Qt's QDateTime
+///
+/// Internally this holds a UniquePtr to a QDateTime which has been constructed on the C++ side.
+pub struct DateTime {
+    // Note that once map_qt_value is removed later, this can become private again
+    #[doc(hidden)]
+    pub(crate) inner: cxx::UniquePtr<QDateTime>,
+}
+
+impl DateTime {
+    /// Construct a Rust DateTime from an existing QDateTime, this is a copy operation.
+    pub fn from_qdatetime(qdatetime: &QDateTime) -> Self {
+        Self {
+            // Safety: TODO
+            inner: unsafe {
+                let mut ptr = MaybeUninit::<cxx::UniquePtr<QDateTime>>::zeroed();
+                qdatetime_init_from_qdatetime(&mut ptr, qdatetime);
+                ptr.assume_init()
+            },
+        }
+    }
+
+    /// Construct a Rust DateTime from a given QDate and QTime
+    pub fn from_date_time(date: &QDate, time: &QTime) -> Self {
+        Self {
+            // Safety: TODO
+            inner: unsafe {
+                let mut ptr = MaybeUninit::<cxx::UniquePtr<QDateTime>>::zeroed();
+                qdatetime_init_from_date_time(&mut ptr, date, time);
+                ptr.assume_init()
+            },
+        }
+    }
+
+    pub fn date(&self) -> QDate {
+        // Safety: TODO
+        unsafe { qdatetime_get_date(&self.inner) }
+    }
+
+    pub fn time(&self) -> QTime {
+        // Safety: TODO
+        unsafe { qdatetime_get_time(&self.inner) }
+    }
+
+    pub fn set_date(&mut self, date: &QDate) {
```

Comment:
More Rust-y way would be to take a `QDate` instead of a `&QDate`, but because the C++ code needs a const reference anyway, I think taking a reference is fine.